plugin.cpp

```diff
@@ -663,6 +689,34 @@ void plugin_ingest(PLUGIN_HANDLE *handle,
 				}
 			}
 		}
+		else if (assetAction->actn == action::SELECT)
+		{
+			Reading *newReading = new Reading(**elem); // copy original Reading object
+                        // Iterate over the datapoints and remove unwanted
+                        vector<Datapoint *>& dps = newReading->getReadingData();
+			vector<string> rmList;
+			for (auto& dp : dps)
+                        {
+				string name = dp->getName();
+				bool found = false;
+                                for (auto& datapoint : assetAction->select)
+					if (datapoint.compare(name) == 0)
+						found = true;
```

Comment:
since assetAction->select is a vector, we can use std::find instead of using for loop

Reply:
We could, but for a short vector like this it is unlikely to make much if any difference.